### ISSUE TITLE
Clarifications on Paper

### ISSUE THREAD

**Issue by altria-zewei-wang** [eric-ai-lab/Soft-Thinking#8]
Dear Team, 

Thank you for your great work, it was indeed a pleasure reading into it. I have a few clarification question on the paper details. It would be greatly appreciated if you can clarify them for me. It has been a bit hard to infer them through codes since it's a modified version of sglang. 

1. I am a bit confused on your terminology when using "probability". When we calculated probability weight mixture of token embeddings, are we taking the (softmaxed logit -> probability of each token) * the embedding given by tokenizer? 

2. Does this mean that we no longer has this textual reasoning chain unless we take the max probability as illustrated in Figure 4 for visualization purpose, but we mainly look at the correctness of the response?

**Comment by namezhenzhang** [comment]
1. Yes. The "probability" is computed using Softmax(logits), where the length corresponds to the vocabulary size, and the new embedding is derived by multiplying the probability with the embedding matrix.  

2. No. We retain the top n tokens in the reasoning chain and visualize it by selecting the token with the maximum probability at each step for clarity. This creates a textual reasoning chain with n tokens per step instead of one. Additionally, the output after </think> does not employ soft thinking and summarizes the reasoning chain, enabling inference of the model's thought process through 1) the top n textual reasoning chain and 2) the output (response) following </think>.

**Comment by WeixiangYAN** [comment]
Hi @altria-zewei-wang , since you haven't responded in the past week, we will do a courtesy check and temporarily close it. If you still have any questions, please feel free to open it and contact us. Thank you for your support of our work!
